otelcol/collector.go

```diff
@@ -66,10 +66,16 @@ type CollectorSettings struct {
 	// and manually handle the signals to shutdown the collector.
 	DisableGracefulShutdown bool
 
+	// Deprecated: [v0.xx.0] Use ConfigProviderSettings instead.
 	// ConfigProvider provides the service configuration.
 	// If the provider watches for configuration change, collector may reload the new configuration upon changes.
 	ConfigProvider ConfigProvider
 
+	// ConfigProviderSettings allows configuring the way the Collector retrieves its configuration
+	// The Collector will reload based on configuration changes from the ConfigProvider if any
+	// confmap.Providers watch for configuration changes.
+	ConfigProviderSettings ConfigProviderSettings
```

Comment:
Before/After doing this:
1. You need to add a new `ConfigConverter` (similar to confmap converter) that allows to change the provided `Config` and before using it. There is an use-case that some have about this.
2. After this is merged we should change `ConfigProvider` to be a struct, no need to be an interface.

Reply:
1. I think this should be covered by https://github.com/open-telemetry/opentelemetry-collector/issues/8940. I'd prefer to to do it after since I expect that will need a few rounds of review.
2. Sounds good.

Reply:
> There is a use-case that some have about this.

Is there a unit test for that use case?